### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by sainingo**
### Project requirements

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.

- [ ] It will return the first user from the database.
- [ ] Don't worry about proper authentication at this time. You will add it in the upcoming projects.

- Create forms to perform the following functions:

      - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
      - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

- Allow Users to add likes to Posts.

**Screenshots**:

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -5,6 +5,36 @@ def index
...
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new
+    render :new, locals: { post: @post }
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    add_post = @user.posts.new(post_params)
```
Comment (codepantha) [inline app/controllers/posts_controller.rb]: - [x] Please note that new posts should be created on behalf of the currently logged-in user (current_user) of the app.
In order to achieve that, you need to do something along the lines of `current_user.posts.new(post_params)`

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def create
+    post = Post.find(params[:id])
+    comment = current_user.comments.new(comment_params)
+    comment.post = post
```
Comment (codepantha) [inline app/controllers/comments_controller.rb]: - [ ] You have a `create` action 👏 but please don't forget to add a `new` action as well to your CommentsController as per requirements 👍 
- [x] When you have done so, please proceed to your `routes.rb` file and specify a **GET** route `/posts/:post_id/comments/new` which will serve up the `views/comments/new.html.erb` view file so users can write comments

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️ 

## Hi @sainingo 👋, 

You have done an amazing job so far 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!



### Highlights ⭐ 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors
✔️ Correct GitFlow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (FrederickMih) [inline app/controllers/posts_controller.rb]: - [ ] As pointed out by the previous reviewer, kindly make sure that a post is created on behalf of the currently logged-in user(current_user) of the app, that's `current_user.posts.new(post_params)`.  You really need to call the `current_user `method previously created in the ApplicationController when creating a post.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @sainingo  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (sainingo) [reply app/controllers/posts_controller.rb]: Hello thank you for the suggestions. I have implemented

Comment (peterrobert) [reply app/controllers/posts_controller.rb]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @sainingo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**